lib/v2/transcriptforest/index.js

```diff
@@ -0,0 +1,76 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const { art } = require('@/utils/render');
+const path = require('path');
+
+module.exports = async (ctx) => {
+    const { channel } = ctx.params;
+    const limit = ctx.query.limit ? Number.parseInt(ctx.query.limit, 10) : 30;
+
+    const rootUrl = 'https://www.transcriptforest.com';
+
+    const { data: firstResponse } = await got(rootUrl);
+
+    const data = JSON.parse(firstResponse.match(/({"props".*"scriptLoader":\[]})<\/script>/)?.[1]);
+
+    const buildId = data.buildId;
+    const defaultLocale = data.defaultLocale;
+    const channels = data.props.pageProps.listChannel;
+    const selected = channel ? channels.find((c) => c.channel_id === channel || c.channel_name === channel) : undefined;
+
+    const apiUrl = new URL(`_next/data/${buildId}/en${selected ? `/channel/${selected.channel_id}` : ''}.json`, rootUrl).href;
+    const currentUrl = new URL(selected ? `${defaultLocale}/channel/${selected.channel_id}` : '', rootUrl).href;
+
+    const { data: response } = await got(apiUrl, {
+        searchParams: {
+            channelName: selected ? selected.channel_id : '',
+            offset: 0,
+        },
+    });
+
+    const items = response.pageProps.listEpisode.data.slice(0, limit).map((item) => ({
+        title: item.episode_name,
+        link: new URL(`${defaultLocale}/${item.channel_id}/${item.episode_id}`, rootUrl).href,
+        description: art(path.join(__dirname, 'templates/description.art'), {
+            audios: [
+                {
+                    src: item.source_media,
```

Comment:
This audio track contains ads.

To retrieve audio track w/o ads, you need to use `pageProps.currentEpisode.media` from `${rootUrl}/_next/data/${buildId}/${defaultLocale}/${item.channel_id}/${item.episode_id}.json` and the url to transcript is `pageProps.currentEpisode.ps4_url`

Reply:
Thanks for the review. 😊